### PR TITLE
docs: Add document on Couler Step Zoo

### PR DESCRIPTION
**Pull request by terrytangyuan**
Signed-off-by: terrytangyuan <terrytangyuan@gmail.com>

### What changes were proposed in this pull request?

Add document on Couler Step Zoo. Related issue https://github.com/couler-proj/couler/issues/67.

cc'ing @couler-proj/couler-team @xinbinhuang @andreyvelich @FogDong @gaocegege @davidhershey @vishnu2kmohan @JPZ13 @inohmonton99 and others who might be interested to take a look and share feedback.

### Why are the changes needed?

This document introduces the Couler Step Zoo, which consists of a collection of pre-defined and reusable steps that can
be used directly as part of a workflow defined using Couler.
### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Not applicable.

### CODE REVIEW

**Comment by xinbinhuang** [comment]
I think it's good for the first draft 👍

**Review by FogDong — COMMENTED** [review]
LGTM

**Review by gaocegege — COMMENTED** [review]
/lgtm

Thanks for your contribution! :tada: :+1:

**Review by ywskycn — APPROVED** [review]

File: docs/couler-step-zoo.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+This document introduces the Couler Step Zoo, which consists of a collection of pre-defined and reusable steps that can
+be used directly as part of a workflow defined using Couler.
+
+
+## Existing Steps
+
+Currently, Couler Step Zoo consists of the following pre-defined steps:
+
+| Name | Description | API |
+| ---- | ----------- | --- |
```
Comment (ywskycn) [inline docs/couler-step-zoo.md:12]: Step may also need to include input output definitions

File: docs/couler-step-zoo.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+1. Make sure the step meets the list of requirements in the above section.
+1. Add the step implementation to [couler.steps module](../couler/steps). The interface would look like the following:
+
+```python
+def random_code():
+    import random
+
+    res = "heads" if random.randint(0, 1) == 0 else "tails"
+    print(res)
```
Comment (Yancey1989) [inline docs/couler-step-zoo.md:45]: `return res` ?

File: docs/couler-step-zoo.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+
+## Requirements on New Steps
+
+In order to provide a consistent and friendly experience to our users, below are the requirements for a new pre-defined
+step to be eligible for inclusion in Couler Step Zoo:
+
+* It should be completely implemented using the core [Couler APIs](couler-api-design.md).
+* It should expose backend specific configurations instead of hard-coding them.
+* It should have a clear set of dependencies that can be easily installed with sufficient instructions.
```
Comment (Yancey1989) [inline docs/couler-step-zoo.md:29]: How to specify the dependencies for a new step function?  Adding the dependencies in [requirements.txt](https://github.com/couler-proj/couler/blob/master/requirements.txt) or other ways?

**Pull request merged**